src/utility/Albany_ThyraCrsMatrixFactory.cpp

```diff
@@ -28,34 +25,35 @@ struct ThyraCrsMatrixFactory::Impl {
 
 ThyraCrsMatrixFactory::
 ThyraCrsMatrixFactory (const Teuchos::RCP<const Thyra_VectorSpace> domain_vs,
-                    const Teuchos::RCP<const Thyra_VectorSpace> range_vs,
-                    const int nonzeros_per_row,
-                    const bool static_profile)
+                       const Teuchos::RCP<const Thyra_VectorSpace> range_vs,
+                       const int /*nonzeros_per_row*/)
  : m_graph(new Impl())
  , m_domain_vs(domain_vs)
  , m_range_vs(range_vs)
  , m_filled (false)
- , m_static_profile (static_profile) 
 {
   auto bt = Albany::build_type();
   TEUCHOS_TEST_FOR_EXCEPTION (bt==BuildType::None, std::logic_error, "Error! No build type set for albany.\n");
 
   if (bt==BuildType::Epetra) {
 #ifdef ALBANY_EPETRA
-    auto e_domain = getEpetraBlockMap(domain_vs);
-    auto e_range  = getEpetraBlockMap(range_vs);
-    m_graph->e_graph = Teuchos::rcp(new Epetra_CrsGraph(Copy,*e_range,nonzeros_per_row,static_profile));
+    e_range  = getEpetraBlockMap(range_vs);
+    e_local_graph.resize(e_range->NumMyElements());
+//    auto e_domain = getEpetraBlockMap(domain_vs);
+//    m_graph->e_graph = Teuchos::rcp(new Epetra_CrsGraph(Copy,*e_range,nonzeros_per_row.data(),true));
 #else
     TEUCHOS_TEST_FOR_EXCEPTION (true, std::logic_error, "Error! Epetra is not enabled in albany.\n");
 #endif
   } else {
-    auto t_domain = getTpetraMap(domain_vs);
-    auto t_range = getTpetraMap(range_vs);
-    if (static_profile) {
-      m_graph->t_graph = Teuchos::rcp(new Tpetra_CrsGraph(t_range,nonzeros_per_row,Tpetra::StaticProfile));
-    } else {
-      m_graph->t_graph = Teuchos::rcp(new Tpetra_CrsGraph(t_range,nonzeros_per_row,Tpetra::DynamicProfile));
-    }
+    t_range = getTpetraMap(range_vs);
+    t_local_graph.resize(t_range->getNodeNumElements());
+
+//    auto t_domain = getTpetraMap(domain_vs);
```

Comment:
IMHO, you should delete all the old code. Keep the files smaller.

Reply:
OK